client/state/sites/domains/actions.js

```diff
@@ -152,33 +151,16 @@ export function disableDomainPrivacy( siteId, domain ) {
 	};
 }
 
-/**
- * @callback onComplete
- * @param {any} error
- * @param {any} data
- * @returns {void}
- */
-
 /**
  * @param {number} siteId
  * @param {string} domain
- * @param {onComplete} onComplete
  */
-export const setPrimaryDomain =
-	( siteId, domain, onComplete = noop ) =>
-	( dispatch ) => {
-		debug( 'setPrimaryDomain', siteId, domain );
-		return wpcom.req.post( `/sites/${ siteId }/domains/primary`, { domain }, ( error, data ) => {
-			if ( error ) {
-				return onComplete( error, data );
-			}
-
-			return dispatch( fetchSiteDomains( siteId ) ).then( () => {
-				onComplete( null, data );
-				dispatch( requestSite( siteId ) );
-			} );
-		} );
-	};
+export const setPrimaryDomain = ( siteId, domain ) => async ( dispatch ) => {
+	debug( 'setPrimaryDomain', siteId, domain );
+	await wpcom.req.post( `/sites/${ siteId }/domains/primary`, { domain } );
+	await dispatch( fetchSiteDomains( siteId ) );
+	await dispatch( requestSite( siteId ) );
```

Comment:
Do these need to happen in sequence or could we use `Promise.all` to await the last in parallel?

Reply:
Good idea, they should run in parallel indeed. Fixed this call and another one in `requestSiteAddressChange`, too.